### ISSUE TITLE
Variable resolution and incorrect warnings in string template expressions in 2023-03

### ISSUE THREAD

**Issue by turchinc** [groovy/groovy-eclipse#1479]
Since updating to 2023-03 with Groovy Development Tools 4.9.0 RELEASE installed through marketplace, I am seeing IMO incorrect errors 
![image](https://github.com/groovy/groovy-eclipse/assets/645266/97bedad7-6b16-4af8-b0ec-0e24d0810a01)
and warnings 
![image](https://github.com/groovy/groovy-eclipse/assets/645266/33f7f9ae-1355-4a0c-a532-9766249693b4)
relating to the resolution of variables in string template expressions. Interestingly, the autocomplete still works as expected providing the objects and methods which then are marked as errors as illustrated in the screenshots. 
The error above is downgraded to a warning (still incorrect) if I simply extract the template expression into a variable:
![image](https://github.com/groovy/groovy-eclipse/assets/645266/ab131636-f3ee-40cc-91b1-5f2dd5c0521f)
It completely disappears if I do not use the curly brackets but simply `$stateKey`
![image](https://github.com/groovy/groovy-eclipse/assets/645266/887f4ac8-6840-40e5-b487-6ed936d1ce52)

This was working in 2022-09 - has something changed or do I need to adjust something on my end?

**Comment by eric-milles** [comment]
Sorry to hear you're having troubles.  This should work and indeed I use GString interpolation often.  I have not seen the "_backtick_ name _backtick_ cannot be resolved to an object" warnings before.  If you look in the Problems view, what is the Type column display?

Can you tell me more about your install?  What is the active Groovy Compiler version?

**Comment by turchinc** [comment]
I had 4.x groovy active but switched back to 3.x, no change
the warnings and errors all have the type "Language Servers":
![image](https://github.com/groovy/groovy-eclipse/assets/645266/450e806c-adea-4f4f-800d-2d250175fb03)

Lemme know if I can provide more feedback thanks!

**Comment by turchinc** [comment]
@eric-milles  thanks! You saved me. It seems that Camel K or some other part of Redhat Tools was causing the problem. I really only need the server runtimes, so I uninstalled the rest of the cruft installed by default in marketplace and now it seems my groovy editor is back to normal:
![image](https://github.com/groovy/groovy-eclipse/assets/645266/f2effee5-d484-4aaf-944d-bf5a220d07c1)

AND I learned about the Language Server settings, which I have never even seen!
